pkg/labels/applicator.go

```diff
@@ -54,6 +54,11 @@ func (l Labeled) SameAs(o Labeled) bool {
 	return l.ID == o.ID && l.LabelType == o.LabelType
 }
 
+type WatchResult struct {
+	Labeled []Labeled
+	Valid   bool // Will be false if the underlying watcher has terminated
```

Comment:
aren't clients going to have to do one or the other anyway to exit their loops?

That said i'm okay with this as a safety mechanism